packages/synapse-react-client/src/components/SynapseTable/datasets/DatasetItemsEditor.tsx

```diff
@@ -140,6 +206,12 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
     useState<RequiredProperties<EntityRefCollectionView, 'items'>>()
   const [previousDatasetToUpdate, setPreviousDatasetToUpdate] =
     useState<RequiredProperties<EntityRefCollectionView, 'items'>>()
+
+  // Stores the `close` function that will be used to hide the last toast that prompted the user to save
+  const [onCloseLastSavePromptToast, setOnCloseLastSavePromptToast] = useState<
+    { close: () => void } | undefined
+  >()
+
```

Comment:
After each change, we push a notification that tells you to save. Change logic to remove the last toast after creating a new one. We store the `close` handler in state here.